CHANGELOG.md

```diff
@@ -1,3 +1,10 @@
+## 0.12.12 (April 7, 2022)
```

Comment:
^^ 0.12.13.  Curious why we reverted instead of making the logic work.

Reply:
Chris, the bug that we ended up reverting was caused by the following: if a setting validator depends on another setting, as is (was) the case with `api_key` and `base_url`, special care must be taken to ensure that certain operations with the `wandb.Settings` object (namely, `__init__`, `copy`, `update`, and `_apply_settings`) are performed in the correct order. https://github.com/wandb/client/pull/3484 does address this particular case in full, but it feels wrong to do it that way in the general case. What if add more validators (or new settings) that add deps on other settings? We have two options in mind:
- defer setting validation to a point in time when all settings from all sources have been gathered.
- construct some sort of a dag that would auto-detect dependencies and ensure ops are performed in the correct order (and also inform the developer if a dep cycle has been introduced into the code).
- there's also an option of simply not allowing setting validators to depend on other settings, (but I'm personally against it).
In all cases, more thought + testing is required, so decided to leave that for the next release and instead get a fix in asap.